server/game/core/cost/CostAdjuster.ts

```diff
@@ -1,37 +1,57 @@
 import type { AbilityContext } from '../ability/AbilityContext';
 import type { IAbilityLimit } from '../ability/AbilityLimit';
 import type { Card } from '../card/Card';
-import { CardType, PlayType, Aspect, CardTypeFilter } from '../Constants';
+import { PlayType, Aspect, CardTypeFilter, AspectFilter } from '../Constants';
 import type Game from '../Game';
 import type Player from '../Player';
 import * as EnumHelpers from '../utils/EnumHelpers';
+import * as Contract from '../../core/utils/Contract';
 
-export enum CostAdjustDirection {
+export enum CostAdjustType {
     Increase = 'increase',
-    Decrease = 'decrease'
+    Decrease = 'decrease',
+    IgnoreAllAspects = 'ignoreAllAspects',
+
+    /** @deprecated Not yet implemented: TODO with Kylo/Rey */
+    IgnoreSpecificAspects = 'ignoreSpecificAspect'
 }
 
 // TODO: refactor so we can add TContext for attachTargetCondition
 export interface ICostAdjusterProperties {
+
+    /** The type of cards that can be reduced */
     cardTypeFilter: CardTypeFilter;
-    amount: number | ((card: Card, player: Player) => number);
-    direction: CostAdjustDirection;
+
+    /** The amount to adjust the cost by */
+    amount?: number | ((card: Card, player: Player) => number);
+
+    /** The type of cost adjustment */
+    costAdjustType: CostAdjustType;
+
+    /** The cost floor (for minimum costs) */
     costFloor?: number;
```

Comment:
@AMMayberry1 Do you think it would make sense to leave it in as deprecated or commented out? I could definitely see a cost floor being a thing at some future point.

Reply:
At the barest minimum I would want to comment it out, as we add more features like PlayForFree we would have to figure out how they would interact with this property and it's not worth it for a feature that doesn't exist yet.

I'm actually a little skeptical it will come in anytime soon, I haven't seen stuff like that a lot (though my tcg experience is limited tbf), so I'm fine with just removing it. Up to you though